lib/spack/spack/reporters/cdash.py

```diff
@@ -107,9 +132,11 @@ def build_report(self, filename, report_data):
             build_pos = phases_encountered.index("build")
             phases_encountered.insert(0, phases_encountered.pop(build_pos))
 
+        self.starttime = self.endtime - total_duration
         for phase in phases_encountered:
+            report_data[phase]['starttime'] = self.starttime
             errors, warnings = parse_log_events(
-                report_data[phase]['log'].splitlines())
+                report_data[phase]['log'].getvalue().splitlines())
```

Comment:
If you're going to just `getvalue().splitlines()` eventually anyway, why even bother with `StringIO`?  Why not just append to an array so you don't have to do anything here?  It's easy to `arr.join('\n')` when you need to, if you need to turn the whole thing into a string.